utensor_cgen/frontend/onnx.py

```diff
@@ -20,13 +20,16 @@
 
 
 def _convert_op_attribute(attrib_pb):
+  ints_attribs = ['perm','dilations','kernel_shape','strides','pads']
   # TODO: integrate with ir.converter
   if attrib_pb.HasField('f'):
     return attrib_pb.f
   elif attrib_pb.HasField('i'):
     return attrib_pb.i
   elif attrib_pb.HasField('s'):
     return attrib_pb.s
+  elif attrib_pb.name in ints_attribs:
```

Comment:
After double-check the `onnx` [proto3 file](https://github.com/onnx/onnx/blob/master/onnx/onnx-ml.proto#L166-L168), I suggest you get the `ints` attribute via `elif attrib_pb.HasFields('ints'): ....` which is more robust than listing.

Reply:
Which will be better if you add `floats` for me XD

Reply:
Added, however HasFields only works with scalar attributes, so a more indirect path was required.

Reply:
You are right.
I don't know `HasField` is not working on repeated field (why protobuf doing this???)

I think maybe it's better to parse it by the `type` attribute:
https://github.com/onnx/onnx/blob/master/onnx/onnx-ml.proto#L154

That is:
```python
....
elif attrib_pb.type == 7:
    # ints
elif attrib_pb.type == 6:
    # floats
....
```

How do you think?

Reply:
Minor cleanup done as follows, as this checks member existence and len > 1:
....
elif attrib_pb.floats:
    # floats
elif attrib_pb.ints:
    # ints
elif attrib_pb.strings:
    # strings
....